diff --git sorc/build_fv3.sh sorc/build_fv3.sh
index 5659b533..b5f3283f 100755
--- sorc/build_fv3.sh
+++ sorc/build_fv3.sh
@@ -18,8 +18,10 @@ fi
 
 if [ $target = hera ]; then target=hera.intel ; fi
 
+if [ "$target" = "linux.intel" ]; then export NEMS_COMPILER=intel ; fi
+
 cd fv3gfs.fd/
 FV3=$( pwd -P )/FV3
 cd tests/
-./compile.sh "$FV3" "$target" "WW3=Y 32BIT=Y" 1
+./compile.sh "$FV3" "$target" "WW3=N 32BIT=Y" 1
 mv -f fv3_1.exe ../NEMS/exe/global_fv3gfs.x
