diff --git CMakeLists.txt CMakeLists.txt
index bfeb9438b..2c48c40bb 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -268,7 +268,7 @@ project(GSI)
     add_subdirectory(src/enkf)
   endif(BUILD_ENKF) 
   if(BUILD_UTIL)
-    add_subdirectory(util/EnKF/gfs/src)
+    #add_subdirectory(util/EnKF/gfs/src)
     add_subdirectory(util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_horiz.fd)
     add_subdirectory(util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_time.fd)
     add_subdirectory(util/Radiance_Monitor/nwprod/radmon_shared.v3.0.0/sorc/verf_radang.fd)
diff --git fix fix
index f0f7447ff..39f80665f 160000
--- fix
+++ fix
@@ -1 +1 @@
-Subproject commit f0f7447ff01d07e7d9b6efe017a62e26541751cb
+Subproject commit 39f80665f0418b9a3a0e9e7564b4151f8ca5bdcf
diff --git libsrc libsrc
index 21f2383e0..7ab5c9b9c 160000
--- libsrc
+++ libsrc
@@ -1 +1 @@
-Subproject commit 21f2383e075a0d0bfd24df60998c061fc4de202a
+Subproject commit 7ab5c9b9c9e022ab4c7a24be555fd959a8645538
diff --git modulefiles/modulefile.ProdGSI.linux modulefiles/modulefile.ProdGSI.linux
new file mode 100644
index 000000000..340be5d68
--- /dev/null
+++ modulefiles/modulefile.ProdGSI.linux
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+#%Module######################################################################
+##
+
+echo "Setting environment variables for gsi on Linux with gcc/gfortran"
+
+##
+## CRTM
+##
+
+export CRTM_VER=2.3.0
+export CRTM_INC=/opt/CRTM-${CRTM_VER}/crtm_v${CRTM_VER}/include
+export CRTM_LIB=/opt/CRTM-${CRTM_VER}/crtm_v${CRTM_VER}/lib/libcrtm.a
diff --git src/fv3gfs_ncio/CMakeLists.txt src/fv3gfs_ncio/CMakeLists.txt
index 337a5378c..007b76ff0 100644
--- src/fv3gfs_ncio/CMakeLists.txt
+++ src/fv3gfs_ncio/CMakeLists.txt
@@ -3,6 +3,7 @@ if(BUILD_FV3GFS_NCIO)
   set(Fortran_MODULE_DIRECTORY "${PROJECT_BINARY_DIR}/include")
   # NetCDF-4 library
   include_directories( ${NETCDF_INCLUDES} ${FV3GFS_NCIO_INCS} )
+  message(STATUS "The value of NETCDF_INCLUDES=${NETCDF_INCLUDES}")
   FILE(GLOB FV3GFS_NCIO_SRC ${CMAKE_CURRENT_SOURCE_DIR}/module_fv3gfs_ncio.f90 )
   set_source_files_properties( ${FV3GFS_NCIO_SRC} PROPERTIES COMPILE_FLAGS ${FV3GFS_NCIO_Fortran_FLAGS} )
   add_library(fv3gfs_ncio STATIC ${FV3GFS_NCIO_SRC})
diff --git src/gsi/phil2.f90 src/gsi/phil2.f90
index 9dce41f21..7701eca24 100644
--- src/gsi/phil2.f90
+++ src/gsi/phil2.f90
@@ -299,7 +299,7 @@ integer(i_kind),parameter         :: ngen=14
 real(dp),       dimension(3,nob)  :: xob
 real(dp),       dimension(nob)    :: rob,sob,latobc
 real(dp),       dimension(0:3)    :: qset
-real(dp),       dimension(0:3,3)  :: qset3
+real(dp),       dimension(0:3,5)  :: qset3
 real(dp),       dimension(0:3,5)  :: qset5
 real(dp),       dimension(0:3,7)  :: qset7
 real(dp),       dimension(0:3,13) :: qset13
@@ -349,7 +349,7 @@ do L=1,nob
    xob(:,L)=(/clat*clon,clat*slon,slon/)
    rob(L)=(altob(L)-vmin)/uv-vrand! <- altitudes in hilbert vertical units
 enddo
-if(nrand<1 .or. nrand>273)stop'nrand is invalid'
+if(nrand<1 .or. nrand>273)stop 'nrand is invalid'
 if(nrand>5)then; call getqset7(      qset7); if(nrand>7)call getqset13(qset13)
 else;            call getqset5(nrand,qset5)
 endif
@@ -407,7 +407,7 @@ do irand=1,nrand
       case(1)
          call bsmoo1(nob,span,sob,rank,wtob)
       case default
-         stop'In denest; this value of B-spline order, nor, is not supported'
+         stop 'In denest; this value of B-spline order, nor, is not supported'
    end select
 enddo! irand
 ! Convert the sum of Hilbert-parameter-relative densities to an average,
@@ -499,7 +499,7 @@ do L=1,nob
    xob(:,L)=(/clat*clon,clat*slon,slon/)
    rob(L)=(altob(L)-vmin)/uv-vrand! <- altitudes in hilbert vertical units
 enddo
-if(nrand<1 .or. nrand>104)stop'nrand is invalid'
+if(nrand<1 .or. nrand>104)stop 'nrand is invalid'
 if(nrand>5)then; call getqset8(      qset8); if(nrand>8)call getqset13(qset13)
 else;            call getqset5(nrand,qset5)
 endif
@@ -549,7 +549,7 @@ do irand=1,nrand
       case(1)
          call bsmoo1(nob,span,sob,rank,denob)
       case default
-         stop'In denest; this value of B-spline order, nor, is not supported'
+         stop 'In denest; this value of B-spline order, nor, is not supported'
    end select
 enddo! irand
 ! Convert the sum of Hilbert-parameter-relative densities to an average,
@@ -650,7 +650,7 @@ do irand=1,nrand
       case(1)
          call bsmoo1(nob,span,sob,rank,wtob)
       case default
-         stop'In denest; this value of B-spline order, nor, is not supported'
+         stop 'In denest; this value of B-spline order, nor, is not supported'
    end select
 ! Rotate the Hilbert tile by a pi/(2*nrand) about its axis at (1/3,1/3):
    if(L<nob)then
@@ -723,7 +723,7 @@ do irand=1,nrand
       case(1)
          call bsmoo1(nob,span,sob,rank,denob)
       case default
-         stop'In denest; this value of B-spline order, nor, is not supported'
+         stop 'In denest; this value of B-spline order, nor, is not supported'
    end select
 ! Rotate the Hilbert tile by a pi/(2*nrand) about its axis at (1/3,1/3):
    if(L<nob)then
@@ -757,7 +757,7 @@ real(dp),parameter:: u8=8.0_dp,or8=u1/sqrt(u8),sig=u1/phi,chi=r2-u1
 real(dp)          :: term1,term2,ce,cf,cg,ch,cj,ck,cl
 !=============================================================================
 qset5(:,1)=(/u1,u0,u0,u0/)
-select case(n); case default; stop'In getqset5; n is outside the valid range'
+select case(n); case default; stop 'In getqset5; n is outside the valid range'
    case(1)
    case(2:3)
       term1=o2+or8
diff --git src/gsi/pvqc.f90 src/gsi/pvqc.f90
index 54ed7b176..81d27ba99 100755
--- src/gsi/pvqc.f90
+++ src/gsi/pvqc.f90
@@ -139,13 +139,13 @@ integer(i_kind)          :: iunit
 logical                  :: ex,op
 !==============================================================================
 if(.not.linitvqc)&
-     stop'In writevqcascfile; VQC parameters and tables are not yet initialized'
-if(npx_a/=npx)stop'In writevqcascfile; mismatched specified npx'
-if(npa_a/=npa)stop'In writevqcascfile; mismatched specified npa'
-if(npb_a/=npb)stop'In writevqcascfile; mismatched specified npb'
-if(npk_a/=npk)stop'In writevqcascfile; mismatched specified npk'
-if(nx_a /=nx )stop'In writevqcascfile; mismatched specified nx'
-if(na_a /=na )stop'In writevqcascfile; mismatched specified na'
+     stop 'In writevqcascfile; VQC parameters and tables are not yet initialized'
+if(npx_a/=npx)stop 'In writevqcascfile; mismatched specified npx'
+if(npa_a/=npa)stop 'In writevqcascfile; mismatched specified npa'
+if(npb_a/=npb)stop 'In writevqcascfile; mismatched specified npb'
+if(npk_a/=npk)stop 'In writevqcascfile; mismatched specified npk'
+if(nx_a /=nx )stop 'In writevqcascfile; mismatched specified nx'
+if(na_a /=na )stop 'In writevqcascfile; mismatched specified na'
 
 do iunit=lunit,nunit
    inquire(unit=iunit, exist=ex, opened=op)
@@ -153,7 +153,7 @@ do iunit=lunit,nunit
    if(.not.op)exit
 enddo
 if(.not.ex .or. iunit>nunit)&
-     stop'In writevqcascfile; No available unit number for writing'
+     stop 'In writevqcascfile; No available unit number for writing'
 open(unit=iunit,file=vqcascfile,access='sequential',form='formatted')
 write(iunit,600)npx,npa,npb,npk,nx,na
 write(iunit,601)sgt
@@ -198,15 +198,15 @@ do iunit=lunit,nunit
    if(.not.op)exit
 enddo
 if(.not.ex .or. iunit>nunit)&
-     stop'In readvqcascfile; No available unit number for reading'
+     stop 'In readvqcascfile; No available unit number for reading'
 open(unit=iunit,file=vqcascfile,access='sequential',form='formatted')
 read(iunit,600)npx,npa,npb,npk,nx,na
-if(npx_a/=npx)stop'In readvqcascfile; mismatched specified npx'
-if(npa_a/=npa)stop'In readvqcascfile; mismatched specified npa'
-if(npb_a/=npb)stop'In readvqcascfile; mismatched specified npb'
-if(npk_a/=npk)stop'In readvqcascfile; mismatched specified npk'
-if(nx_a /=nx )stop'In readvqcascfile; mismatched specified nx'
-if(na_a /=na )stop'In readvqcascfile; mismatched specified na'
+if(npx_a/=npx)stop 'In readvqcascfile; mismatched specified npx'
+if(npa_a/=npa)stop 'In readvqcascfile; mismatched specified npa'
+if(npb_a/=npb)stop 'In readvqcascfile; mismatched specified npb'
+if(npk_a/=npk)stop 'In readvqcascfile; mismatched specified npk'
+if(nx_a /=nx )stop 'In readvqcascfile; mismatched specified nx'
+if(na_a /=na )stop 'In readvqcascfile; mismatched specified na'
 nkm=npk-1
 npb2=npb*2
 allocate(sgt(-nx:nx,0:na,-nkm:nkm),swt(-nx:nx,0:na,-nkm:nkm),&
@@ -250,13 +250,13 @@ integer(i_kind)          :: iunit
 logical          :: ex,op
 !==============================================================================
 if(.not.linitvqc)&
-     stop'In writevqcdatfile; VQC parameters and tables are not yet initialized'
-if(npx_a/=npx)stop'In writevqcdatfile; mismatched specified npx'
-if(npa_a/=npa)stop'In writevqcdatfile; mismatched specified npa'
-if(npb_a/=npb)stop'In writevqcdatfile; mismatched specified npb'
-if(npk_a/=npk)stop'In writevqcdatfile; mismatched specified npk'
-if(nx_a /=nx )stop'In writevqcdatfile; mismatched specified nx'
-if(na_a /=na )stop'In writevqcdatfile; mismatched specified na'
+     stop 'In writevqcdatfile; VQC parameters and tables are not yet initialized'
+if(npx_a/=npx)stop 'In writevqcdatfile; mismatched specified npx'
+if(npa_a/=npa)stop 'In writevqcdatfile; mismatched specified npa'
+if(npb_a/=npb)stop 'In writevqcdatfile; mismatched specified npb'
+if(npk_a/=npk)stop 'In writevqcdatfile; mismatched specified npk'
+if(nx_a /=nx )stop 'In writevqcdatfile; mismatched specified nx'
+if(na_a /=na )stop 'In writevqcdatfile; mismatched specified na'
 
 do iunit=lunit,nunit
    inquire(unit=iunit, exist=ex, opened=op)
@@ -264,7 +264,7 @@ do iunit=lunit,nunit
    if(.not.op)exit
 enddo
 if(.not.ex .or. iunit>nunit)&
-     stop'In writevqcdatfile; No available unit number for writing'
+     stop 'In writevqcdatfile; No available unit number for writing'
 open(unit=iunit,file=vqcdatfile,access='sequential',form='unformatted')
 write(unit=iunit)npx,npa,npb,npk,nx,na
 write(iunit)sgt
@@ -306,15 +306,15 @@ do iunit=lunit,nunit
    if(.not.op)exit
 enddo
 if(.not.ex .or. iunit>nunit)&
-     stop'In readvqcdatfile; No available unit number for reading'
+     stop 'In readvqcdatfile; No available unit number for reading'
 open(unit=iunit,file=vqcdatfile,access='sequential',form='unformatted')
 read(iunit)npx,npa,npb,npk,nx,na
-if(npx_a/=npx)stop'In readvqcdatfile; mismatched specified npx'
-if(npa_a/=npa)stop'In readvqcdatfile; mismatched specified npa'
-if(npb_a/=npb)stop'In readvqcdatfile; mismatched specified npb'
-if(npk_a/=npk)stop'In readvqcdatfile; mismatched specified npk'
-if(nx_a /=nx )stop'In readvqcdatfile; mismatched specified nx'
-if(na_a /=na )stop'In readvqcdatfile; mismatched specified na'
+if(npx_a/=npx)stop 'In readvqcdatfile; mismatched specified npx'
+if(npa_a/=npa)stop 'In readvqcdatfile; mismatched specified npa'
+if(npb_a/=npb)stop 'In readvqcdatfile; mismatched specified npb'
+if(npk_a/=npk)stop 'In readvqcdatfile; mismatched specified npk'
+if(nx_a /=nx )stop 'In readvqcdatfile; mismatched specified nx'
+if(na_a /=na )stop 'In readvqcdatfile; mismatched specified na'
 nkm=npk-1
 npb2=npb*2
 allocate(sgt(-nx:nx,0:na,-nkm:nkm),swt(-nx:nx,0:na,-nkm:nkm),&
@@ -358,13 +358,13 @@ real(dp)          :: bc,p,q,qx,sx,alpha,beta,kappa, &
                      x1,x2,xa,ya,xx
 integer(i_kind)           :: ja
 !==============================================================================
-if(.not.linitvqc)stop'In vqch; VQC tables are not initialized'
+if(.not.linitvqc)stop 'In vqch; VQC tables are not initialized'
 if(ia<0)then; sx=-x; ja=-ia
 else;         sx= x; ja= ia
 endif
-if(ja>na              )stop'In vqch; ia out of bounds'
-if(ib<=0.or.ib>=npb2  )stop'In vqch; ib out of bounds'
-if(ik<=-npk.or.ik>=npk)stop'In vqch; ik out of bounds'
+if(ja>na              )stop 'In vqch; ia out of bounds'
+if(ib<=0.or.ib>=npb2  )stop 'In vqch; ib out of bounds'
+if(ik<=-npk.or.ik>=npk)stop 'In vqch; ik out of bounds'
 x1=x1t(ja,ik)
 x2=x2t(ja,ik)
 xa=xat(ja,ik)
@@ -428,8 +428,8 @@ real(dp),intent(out):: g,w
 real(dp),parameter:: pio4=pi/4_dp
 real(dp)          :: bc,p,q,qx,x1,x2,ya,xx
 !==============================================================================
-if(.not.linitvqc)stop'In vqch; VQC tables are not initialized'
-if(beta<=u0.or.beta>=u2)stop'In vqch; beta out of bounds'
+if(.not.linitvqc)stop 'In vqch; VQC tables are not initialized'
+if(beta<=u0.or.beta>=u2)stop 'In vqch; beta out of bounds'
 x1=x1t(0,0)
 x2=x2t(0,0)
 ya=yat(0,0)
@@ -478,13 +478,13 @@ real(dp)          :: bc,p,q,qx,sx,w1,w2,xodx,beta,kappa,xe,ge,we,&
                      ww,dfa,fl,dfl,f1,f2,df1,df2,g1,g2
 integer(i_kind)   :: ix1,ix2,ja
 !==============================================================================
-if(.not.linitvqc)stop'In vqcs; VQC tables are not initialized'
+if(.not.linitvqc)stop 'In vqcs; VQC tables are not initialized'
 if(ia<0)then; sx=-x; ja=-ia
 else;         sx= x; ja= ia
 endif
-if(ja>na              )stop'In vqcs; ia out of bounds'
-if(ib<=0.or.ib>=npb2  )stop'In vqcs; ib out of bounds'
-if(ik<=-npk.or.ik>=npk)stop'In vqcs; ik out of bounds'
+if(ja>na              )stop 'In vqcs; ia out of bounds'
+if(ib<=0.or.ib>=npb2  )stop 'In vqcs; ib out of bounds'
+if(ik<=-npk.or.ik>=npk)stop 'In vqcs; ik out of bounds'
 beta =ib*db
 kappa=ik*dk
 bc=tan(pio4*(u2-beta))
@@ -574,7 +574,7 @@ real(dp)          :: bc,p,q,qx,w1,w2,xodx,xe,ge,we,&
                      ww,dfa,fl,dfl,f1,f2,df1,df2,g1,g2
 integer(i_kind)   :: ix1,ix2
 !==============================================================================
-if(.not.linitvqc)stop'In vqcs; VQC tables are not initialized'
+if(.not.linitvqc)stop 'In vqcs; VQC tables are not initialized'
 bc=tan(pio4*(u2-beta))
 p=bc**2
 q=u1/bc
diff --git src/ncdiag/CMakeLists.txt src/ncdiag/CMakeLists.txt
index 154a098ed..0f0a412be 100644
--- src/ncdiag/CMakeLists.txt
+++ src/ncdiag/CMakeLists.txt
@@ -8,6 +8,7 @@ if(BUILD_NCDIAG)
   # NetCDF-4 library
   include_directories( ${NETCDF_INCLUDES} ${NCDIAG_INCS} )
 
+  message( STATUS "NETCDF_LIBRARIES=${NETCDF_LIBRARIES} NETCDF_INCLUDES=${NETCDF_INCLUDES}")
   # 32-bit reals, for now
   add_definitions(-D_REAL4_ -DUSE_MPI)
   FILE(GLOB NCDIAG_SRC ${CMAKE_CURRENT_SOURCE_DIR}/*90 )
diff --git ush/build_all_cmake.sh ush/build_all_cmake.sh
index ceaf91d9c..9931ac12a 100755
--- ush/build_all_cmake.sh
+++ ush/build_all_cmake.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 set -ex
 
@@ -43,6 +43,8 @@ elif [[ -d /discover ]] ; then
 elif [[ -d /work ]]; then
     . $MODULESHOME/init/sh
     target=orion
+elif [[ -d /dev ]] ; then
+    target=linux
 else
     echo "unknown target = $target"
     exit 9
diff --git util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_horiz.fd/read_diag.f90 util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_horiz.fd/read_diag.f90
index 7e592d3a5..774d0a05e 100644
--- util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_horiz.fd/read_diag.f90
+++ util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_horiz.fd/read_diag.f90
@@ -649,7 +649,7 @@ module read_diag
   
     nlevs = header_fix%nlevs 
 
-    if( ncdiag_open_status(cur_idx)%nc_read == .true. ) then 
+    if( ncdiag_open_status(cur_idx)%nc_read .eqv. .true. ) then 
        iflag = -1  
     else  
        iflag = 0
@@ -663,7 +663,8 @@ module read_diag
           ntobs = nrecords
        end if
 
-       write(6,*) 'header_fix%nlevs, ncdiag_open_status(id)%num_records, ntobs = ', header_fix%nlevs, ncdiag_open_status(id)%num_records, ntobs
+       write(6,*) 'header_fix%nlevs, ncdiag_open_status(id)%num_records, ntobs = ', &
+               header_fix%nlevs, ncdiag_open_status(id)%num_records, ntobs
 
        !------------------------------------
        !  allocate the returned structures
@@ -954,7 +955,7 @@ module read_diag
         end if 
      end do 
 
-     if( verify_var_name_nc == .false. ) then
+     if( verify_var_name_nc .eqv. .false. ) then
         do k=1,num_vars
            if( test_name == var_names(k) ) then
               verify_var_name_nc = .true.
diff --git util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_time.fd/read_diag.f90 util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_time.fd/read_diag.f90
index 7e592d3a5..774d0a05e 100644
--- util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_time.fd/read_diag.f90
+++ util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_time.fd/read_diag.f90
@@ -649,7 +649,7 @@ module read_diag
   
     nlevs = header_fix%nlevs 
 
-    if( ncdiag_open_status(cur_idx)%nc_read == .true. ) then 
+    if( ncdiag_open_status(cur_idx)%nc_read .eqv. .true. ) then 
        iflag = -1  
     else  
        iflag = 0
@@ -663,7 +663,8 @@ module read_diag
           ntobs = nrecords
        end if
 
-       write(6,*) 'header_fix%nlevs, ncdiag_open_status(id)%num_records, ntobs = ', header_fix%nlevs, ncdiag_open_status(id)%num_records, ntobs
+       write(6,*) 'header_fix%nlevs, ncdiag_open_status(id)%num_records, ntobs = ', &
+               header_fix%nlevs, ncdiag_open_status(id)%num_records, ntobs
 
        !------------------------------------
        !  allocate the returned structures
@@ -954,7 +955,7 @@ module read_diag
         end if 
      end do 
 
-     if( verify_var_name_nc == .false. ) then
+     if( verify_var_name_nc .eqv. .false. ) then
         do k=1,num_vars
            if( test_name == var_names(k) ) then
               verify_var_name_nc = .true.
diff --git util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_time.fd/time.f90 util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_time.fd/time.f90
index d6402196a..8ee11d325 100755
--- util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_time.fd/time.f90
+++ util/Ozone_Monitor/nwprod/oznmon_shared.v2.0.0/sorc/oznmon_time.fd/time.f90
@@ -330,7 +330,7 @@ program main
    end do
 
 !  Do validation
-   if( validate == .TRUE. ) then
+   if( validate .eqv. .TRUE. ) then
       call load_base( satname, ier )
 
       write(6,*) 'ier from load_base = ', ier
@@ -345,7 +345,7 @@ program main
    do j=1,n_levs
       if ( use(j,k) > 0.0 ) then
 
-         if( validate == .TRUE. .AND. ier >= 0 ) then
+         if( validate .eqv. .TRUE. .AND. ier >= 0 ) then
 
             pbound = 0.00
             call validate_penalty( j, k, penalty(j,k), valid_penalty, pbound, iret )
@@ -370,7 +370,7 @@ program main
       endif
    end do
 
-   if( validate == .TRUE. ) then
+   if( validate .eqv. .TRUE. ) then
       close( lupen )
       close( lucnt )
    endif
