diff --git fv3_cap.F90 fv3_cap.F90
index b9162a8..30bd783 100644
--- fv3_cap.F90
+++ fv3_cap.F90
@@ -1118,7 +1118,7 @@ module fv3gfs_cap_mod
          timerhi = mpi_wtime()
 !         if (mype == 0 .or. mype == lead_wrttask(1)) print *,' aft fcst run alarm is on, na=',na,'mype=',mype
 
-         call ESMF_VMEpochEnter(epoch=ESMF_VMEpoch_Buffer, rc=rc)
+         !call ESMF_VMEpochEnter(epoch=ESMF_VMEpoch_Buffer, rc=rc)
          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
          do i=1, FBCount
 !
@@ -1131,7 +1131,7 @@ module fv3gfs_cap_mod
 !
 !end FBcount
          enddo
-         call ESMF_VMEpochExit(rc=rc)
+         !call ESMF_VMEpochExit(rc=rc)
          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
 
          if (mype == 0 .or. mype == lead_wrttask(n_group).or.mype==last_wrttask(n_group)) print *,'aft fldbdlregrid,na=',na,  &
@@ -1412,7 +1412,7 @@ module fv3gfs_cap_mod
        output: IF(lalarm .or. na==first_kdt ) then
 
          timerhi = mpi_wtime()
-         call ESMF_VMEpochEnter(epoch=ESMF_VMEpoch_Buffer, rc=rc)
+         !call ESMF_VMEpochEnter(epoch=ESMF_VMEpoch_Buffer, rc=rc)
          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
          do i=1, FBCount
 !
@@ -1425,7 +1425,7 @@ module fv3gfs_cap_mod
 !
 !end FBcount
          enddo
-         call ESMF_VMEpochExit(rc=rc)
+         !call ESMF_VMEpochExit(rc=rc)
          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
          if (mype == 0 .or. mype == lead_wrttask(n_group)) print *,'aft fieldbundleregrid,na=',na,  &
                                                                    ' time=', timerh- timerhi
